gpflow/optimizers/mcmc.py

```diff
@@ -46,15 +43,18 @@ def run_chain_fn():
 
         hmc_samples = run_chain_fn()
         parameter_samples = hmc_helper.convert_samples_to_parameter_values(hmc_samples)
-
-    :param target_log_prob_fn: a callable which represents log-density under
-        the target distribution.
-    :param parameters: List of :class:`gpflow.Parameter` used as a state of the Markov chain.
     """
 
-    def __init__(self, target_log_prob_fn: LogProbabilityFunction, parameters: Parameters):
-        if not all([isinstance(p, Parameter) and p.prior is not None for p in parameters]):
```

Comment:
as it's wrapped in `all`, can simply use the generator form instead of the list comprehension